libs/server/Storage/Functions/ObjectStore/ReadMethods.cs

```diff
@@ -24,63 +32,36 @@ public bool SingleReader(ref byte[] key, ref ObjectInput input, ref IGarnetObjec
+                    var opResult = ((IGarnetObject)srcLogRecord.ValueObject).Operate(ref input, ref output, functionsState.respProtocolVersion, out _);
+                    if (output.HasWrongType)
                         return true;
 
-                    case GarnetObjectType.ExpireTime:
-                        var expireTime = ConvertUtils.UnixTimeInSecondsFromTicks(value.Expiration > 0 ? value.Expiration : -1);
-                        CopyRespNumber(expireTime, ref dst.SpanByteAndMemory);
-                        return true;
-                    case GarnetObjectType.PExpireTime:
-                        expireTime = ConvertUtils.UnixTimeInMillisecondsFromTicks(value.Expiration > 0 ? value.Expiration : -1);
-                        CopyRespNumber(expireTime, ref dst.SpanByteAndMemory);
-                        return true;
-
-                    default:
-                        if ((byte)input.header.type < CustomCommandManager.CustomTypeIdStartOffset)
-                        {
-                            var opResult = value.Operate(ref input, ref dst, functionsState.respProtocolVersion, out _);
-                            if (dst.HasWrongType)
-                                return true;
-
-                            return opResult;
-                        }
-
-                        if (IncorrectObjectType(ref input, value, ref dst.SpanByteAndMemory))
-                        {
-                            dst.OutputFlags |= ObjectStoreOutputFlags.WrongType;
-                            return true;
-                        }
+                    return opResult;
+                }
 
-                        var customObjectCommand = GetCustomObjectCommand(ref input, input.header.type);
-                        var writer = new RespMemoryWriter(functionsState.respProtocolVersion, ref dst.SpanByteAndMemory);
-                        try
-                        {
-                            var result = customObjectCommand.Reader(key, ref input, value, ref writer, ref readInfo);
-                            return result;
-                        }
-                        finally
-                        {
-                            writer.Dispose();
-                        }
+                if (IncorrectObjectType(ref input, (IGarnetObject)srcLogRecord.ValueObject, ref output.SpanByteAndMemory))
+                {
+                    output.OutputFlags |= OutputFlags.WrongType;
+                    return true;
+                }
 
+                var customObjectCommand = GetCustomObjectCommand(ref input, input.header.type);
+                var writer = new RespMemoryWriter(functionsState.respProtocolVersion, ref output.SpanByteAndMemory);
+                try
+                {
+                    var result = customObjectCommand.Reader(srcLogRecord.Key, ref input, Unsafe.As<IGarnetObject>(srcLogRecord.ValueObject), ref writer, ref readInfo);
```

Comment:
```suggestion
                    var result = customObjectCommand.Reader(srcLogRecord.Key, ref input, (IGarnetObject>)srcLogRecord.ValueObject, ref writer, ref readInfo);
```

or we cast at the start of the branch and avoid casting for three times.